### PR TITLE
Decrease select_with_timeout to fix 520 error

### PR DESCRIPTION
**Pull request by NathanFlurry**

### CODE REVIEW

**Comment by NathanFlurry** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #287** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/287?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #288** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/288?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/rivet/287?utm_source=stack-comment).

File: lib/util/macros/src/lib.rs [hunk 1]
```diff
@@ -9,7 +9,7 @@ use syn::token::Return;
 use syn::{braced, bracketed, Expr, Ident, Result, Token};
 
 // 1 min
-const DEFAULT_TIMEOUT: u64 = 60 * 1000;
+const DEFAULT_TIMEOUT: u64 = 45 * 1000;
```
Comment (AngelOnFira) [inline lib/util/macros/src/lib.rs:12]: It's a bit hard to know what this magic number means when it's beside the doc saying "1 min", is there a way to make this a bit more clear? Maybe even having a link to the new API_TIMEOUTS.md doc.

File: infra/tf/k8s_infra/traefik.tf [hunk 2]
```diff
@@ -61,9 +61,9 @@ resource "helm_release" "traefik" {
 		additionalArguments = [
 			"--providers.http.endpoint=http://rivet-api-route.rivet-service.svc.cluster.local/traefik/config/core?token=${module.traefik_secrets.values["rivet/api_route/token"]}",
 			"--providers.http.pollInterval=2.5s",
-			# 60s for the long polling requests to gracefully exit + 30s for padding
-			"--entryPoints.web.transport.lifeCycle.graceTimeOut=90s",
-			"--entryPoints.websecure.transport.lifeCycle.graceTimeOut=90s",
+			# 60s for the long polling requests to gracefully exit + 15s grace period
+			"--entryPoints.web.transport.lifeCycle.graceTimeOut=45s",
+			"--entryPoints.websecure.transport.lifeCycle.graceTimeOut=45s",
```
Comment (AngelOnFira) [inline infra/tf/k8s_infra/traefik.tf:66]: Not sure if the math here is correct, wouldn't an extra 15s grace period put it up to `75s`?

File: docs/infrastructure/API_TIMEOUTS.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# API Timeouts
+
+We assume that all requests should take less than. Many load balancers have 60s configured as default timeout.
```
Comment (AngelOnFira) [inline docs/infrastructure/API_TIMEOUTS.md:3]: Missing end of sentence:

> We assume that all requests should take less than...

**Review by AngelOnFira — CHANGES_REQUESTED** [review]
Some clarity requests

**Review by MasterPtato — COMMENTED** [review]
What exactly caused the 520 and why was decreasing the timeout to 40s a fix? Was it traefik's 60 second timeout? Shouldnt we be able to configure that? I don't understand from the docs

**Comment by NathanFlurry** [comment]
> What exactly caused the 520 and why was decreasing the timeout to 40s a fix? Was it traefik's 60 second timeout? Shouldnt we be able to configure that? I don't understand from the docs

This didn't fix the 520, but it's still good to clean it up. I'm fairly certain it's something between CF & AWS.

**Review by AngelOnFira — APPROVED** [review]

**Pull request merged**